### PR TITLE
[PodSecurity] hostProcess baseline check

### LINKED ISSUE #103381: [PodSecurity] baseline - hostProcess check
Implement the "hostProcess" check from https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#pod-security-standards

- [ ] implement check_hostProcess.go for the baseline policy
  - fail if `spec.securityContext.windowsOptions.hostProcess=true` or any containers set `securityContext.windowsOptions.hostProcess=true`, include the relevant container names in the forbidden details
  - see staging/src/k8s.io/pod-security-admission/policy/check_runAsNonRoot.go as an example
- [ ] implement fixtures_hostProcess.go
  - see staging/src/k8s.io/pod-security-admission/test/fixtures_runAsNonRoot.go as an example
  - generatePass should return pods with containers that don't set hostProcess (and which do set hostProcess to false? might need a TODO to wait until the feature is GA)
  - generateFail should generate test cases (will need to mark the `WindowsHostProcessContainers` feature as required for the failure cases)
    - all containers set hostProcess to `true`
    - podspec sets hostProcess to `true`
  - run `UPDATE_POD_SECURITY_FIXTURE_DATA=true go test k8s.io/pod-security-admission/test` to update test fixtures
  - run `go test ./test/integration/auth -v -run PodSecurity` to ensure integration tests pass

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubernetes/issues/103381

Even though hostProcess is alpha, we can still forbid `true` values in the field in the baseline policy

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @tallclair @marosset 
/sig auth windows

### CODE REVIEW

File: staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+				}),
+			}
+		},
+	}
+
+	registerFixtureGenerator(
+		fixtureKey{level: api.LevelBaseline, version: api.MajorMinorVersion(1, 0), check: "hostProcess"},
+		fixtureData_1_0,
+	)
+	// TODO: register another set of fixtures with passing test cases that explicitly set hostProcess=false at pod and container level once hostProcess is GA
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go:76]: Can't we add these tests now? They should pass either way (even if the field is dropped).

File: staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		*podSpec.SecurityContext.WindowsOptions.HostProcess {
+		podSpecForbidden = true
+	}
+
+	// pod or containers explicitly set hostProcess=true
+	if len(forbiddenContainers) > 0 || podSpecForbidden {
+		return CheckResult{
+			Allowed:         false,
+			ForbiddenReason: "hostProcess == true",
+			ForbiddenDetail: strings.Join(forbiddenContainers, ", "),
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go:82]: I'm still not sure about these error messages, but let's punt to the cleanup PR. For the record, I'd suggest something like this:
```suggestion
			ForbiddenReason: "windows hostProcess",
			// No ForbiddenDetail for boolean fields
```

File: staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			if p.Spec.Containers[0].SecurityContext.WindowsOptions == nil {
+				p.Spec.Containers[0].SecurityContext.WindowsOptions = &corev1.WindowsSecurityContextOptions{}
+			}
+			if p.Spec.InitContainers[0].SecurityContext.WindowsOptions == nil {
+				p.Spec.InitContainers[0].SecurityContext.WindowsOptions = &corev1.WindowsSecurityContextOptions{}
+			}
+			return []*corev1.Pod{
+				// true for pod
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.HostNetwork = true
```
Comment (tallclair) [inline staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go:59]: Add a comment here about why HostNetwork must be set.

**Comment by immuzz** [comment]
cc : @jsturtevant @immuzz

File: staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go:76]: requests trying to explicitly set hostProcess fields on creation get rejected when the feature is off; discussed with sig-windows and were hard-pressed to think of a way that dropping the data and failing at runtime improved the user experience. discussion at https://github.com/kubernetes/kubernetes/pull/99576#issuecomment-832259530

File: staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go:82]: I'm fine with reworking these... will link to cleanup issue

File: staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go (hunk 3 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/test/fixtures_hostProcess.go:59]: done

File: staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/policy/check_hostProcess.go:82]: linked

**Pull request merged**